LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DeviceLoaderSynchronizer.cs

```diff
@@ -0,0 +1,330 @@
+                        {
+                            this.AddToDeviceQueue(device, request);
+                            requestHandled = true;
+                            break;
+                        }
+                    }
+                }
+
+                if (!requestHandled)
+                {
+                    var failedReason = hasDevicesMatchingDevAddr ? LoRaDeviceRequestFailedReason.NotMatchingDeviceByMicCheck : LoRaDeviceRequestFailedReason.NotMatchingDeviceByDevAddr;
+                    this.LogRequestFailed(request, failedReason);
+                    request.NotifyFailed(null, failedReason);
+                }
+            }
+
+            this.queuedRequests.Clear();
+        }
+
+        public void Queue(LoRaRequest request)
+        {
+            var requestAddedToQueue = false;
+            if (this.state != LoaderState.Finished)
+            {
+                lock (this.queueLock)
+                {
+                    // add to the queue only if loading is not yet finished
+                    if (this.state < LoaderState.DispatchingQueuedItems)
+                    {
+                        this.queuedRequests.Add(request);
+                        requestAddedToQueue = true;
+                        Logger.Log(this.devAddr, $"Request added to queue. Loader state: {this.state}", LogLevel.Debug);
+                    }
+                }
+            }
+
+            if (!requestAddedToQueue)
+            {
+                Logger.Log(this.devAddr, $"Will try to queue request directly into device. Loader state: {this.state}", LogLevel.Debug);
+
+                foreach (var device in this.destinationDictionary.Values)
+                {
+                    if (request.Payload.CheckMic(device.NwkSKey))
+                    {
+                        this.AddToDeviceQueue(device, request);
+                        return;
+                    }
+                }
+
+                // not handled, raised failed event
+                var failedReason =
+                    this.loadingDevicesFailed ? LoRaDeviceRequestFailedReason.ApplicationError :
+                    this.destinationDictionary.Count > 0 ? LoRaDeviceRequestFailedReason.NotMatchingDeviceByMicCheck : LoRaDeviceRequestFailedReason.NotMatchingDeviceByDevAddr;
+
+                this.LogRequestFailed(request, failedReason);
+
+                request.NotifyFailed(null, failedReason);
+            }
+        }
+
+        private void AddToDeviceQueue(LoRaDevice device, LoRaRequest request)
+        {
+            if (device.IsOurDevice)
+            {
+                device.Queue(request);
+            }
+            else
+            {
+                Logger.Log(device.DevEUI, $"device is not our device, ignore message", LogLevel.Debug);
+                request.NotifyFailed(device, LoRaDeviceRequestFailedReason.BelongsToAnotherGateway);
+            }
+        }
+
+        void SetState(LoaderState newState)
```

Comment:
why not use a property here?

Reply:
Personal preference. If I am doing a bit of code in the setter (logging and validation) I prefer methods.

Reply:
I thought there was a stylecop rule that would flag set/get naming. No big deal.